### PR TITLE
Fix email

### PR DESCRIPTION
**Pull request by koheiw**
Current regex for email is too simple.

### CODE REVIEW

File: R/tokenizers.R [hunk 1]
```diff
@@ -160,7 +160,7 @@ tokenize_word4 <- function(x, split_hyphens = FALSE, split_tags = FALSE, split_e
     
     ftp <- "s?ftp://[-+a-zA-Z0-9@#:.%~=_&/]+"
     http <- "(https?://)?(www.)?[-a-zA-Z0-9]+(\\.[-a-zA-Z0-9]+)+([/?#][-+a-zA-Z0-9@#:.%~=_&]+)*[/?#]?"
-    email <- "[A-Za-z0-9_]+@[A-Za-z][A-Za-z0-9_]+\\.[a-z]+"
+    email <- "[-a-zA-Z0-9_.]+@[-a-zA-Z0-9]+(\\.[-a-zA-Z0-9]+)*\\.[a-z]{2,4}"
```
Comment (odelmarcelle) [inline R/tokenizers.R:163]: I would perhaps drop the quantifier at the end. I looked it up and domains longer than 4 character such as .london or .tokyo exists.

Also, the leading part of the email can sometimes accept '+'. I would add it to the first character set.

Comment (koheiw) [reply R/tokenizers.R]: Thanks. I updated accordingly.

Comment (kbenoit) [reply R/tokenizers.R]: Here are some nightmare scenarios but pretty rare. I can't see an easy way to deal with the spaces issue, so I propose we just ignore these. (The one starting with an underscore works.)
https://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/

**Review by kbenoit — APPROVED** [review]
Would be good to merge this into v3_3 as well, before I send that update to CRAN later this week.

**Pull request merged**